### PR TITLE
remove onnx internal ir code

### PR DESCRIPTION
**Pull request by liqunfu**
### Description
initial code to remove onnx internal ir code.

### Motivation and Context
internal IR code is only used by version converter. I hope by removing it, version converter will become simpler.

### CODE REVIEW

**Review by jcwchen — COMMENTED** [review]
IIRC, https://github.com/onnx/optimizer is also using some of onnx/common/ir.h. It's better to let them know and review as well.

cc @daquexian

File: onnx/common/ir.h [hunk 1]
```diff
@@ -284,7 +284,8 @@ static inline bool operator==(const Use& a, const Use& b) {
 using node_list = std::vector<Node*>;
 using value_list = std::vector<Value*>;
 using use_list = std::vector<Use>;
-using NodeKind = Symbol;
+//using NodeKind = Symbol;
```
Comment (justinchuby) [inline onnx/common/ir.h:287]: nit: Dead code

File: onnx/version_converter/adapters/axes_attribute_to_input.h [hunk 2]
```diff
@@ -16,27 +17,28 @@ class AxesAttributeToInput : public Adapter {
...
-      data.emplace_back(a);
-    }
-
-    Node* constant = graph->create(kConstant);
-    constant->insertBefore(node);
-    constant->t_(kvalue, t);
-    node->addInput(constant->output());
+  void attrToInput(GraphProto* graph_proto, NodeProto* node_proto, std::vector<int64_t>& axes) const {
+    node_proto->attribute();
+    const std::string constant_node_name("aaa");
```
Comment (justinchuby) [inline onnx/version_converter/adapters/axes_attribute_to_input.h:22]: Is this `"aaa"` thing intended? If so a comment may help

File: onnx/version_converter/adapters/axes_attribute_to_input.h [hunk 3]
```diff
@@ -16,27 +17,28 @@ class AxesAttributeToInput : public Adapter {
...
-    auto& data = t.int64s();
-    for (auto a : axes) {
-      data.emplace_back(a);
-    }
-
-    Node* constant = graph->create(kConstant);
-    constant->insertBefore(node);
-    constant->t_(kvalue, t);
-    node->addInput(constant->output());
+  void attrToInput(GraphProto* graph_proto, NodeProto* node_proto, std::vector<int64_t>& axes) const {
```
Comment (justinchuby) [inline onnx/version_converter/adapters/axes_attribute_to_input.h:20]: Just out of curiosity: is it a convention to pass protos by pointer?

File: onnx/version_converter/adapters/broadcast_backward_compatibility.h [hunk 4]
```diff
@@ -16,35 +16,42 @@ class BroadcastBackwardCompatibility final : public Adapter {
...
         "not have broadcastable inputs.",
         name().c_str(),
         initial_version().version(),
         target_version().version());
+
     if (req_broadcast == 1) {
       // If conditional is not fulfilled, we have a default broadcast
       // Add broadcast attribute
-      node->i_(kbroadcast, 1);
+      *node->add_attribute() = MakeAttribute("aa", (int64_t)1);
```
Comment (justinchuby) [inline onnx/version_converter/adapters/broadcast_backward_compatibility.h:50]: What is "aa"?

**Review by justinchuby — APPROVED** [review]

File: onnx/version_converter/adapters/axes_attribute_to_input.h (hunk 2 above)
Comment (liqunfu) [reply onnx/version_converter/adapters/axes_attribute_to_input.h:22]: I will clean up the code when it is clear that the approach is working. Sorry for the confusion.

File: onnx/version_converter/adapters/axes_attribute_to_input.h (hunk 3 above)
Comment (liqunfu) [reply onnx/version_converter/adapters/axes_attribute_to_input.h:20]: good point! previously it was passing shared_ptr. It required the variable was a shared_ptr at calling site so it had to be always not a nullptr. To pass as reference is not good. I am using pointer, which is not as reliable. I will change it back to reference.

File: onnx/common/ir.h [hunk 5]
```diff
@@ -67,10 +67,10 @@ class ResourceGuard final {
   }
 };
 
-struct Dimension final {
-  Dimension() : is_unknown(true), is_int(false), dim(-1) {}
-  Dimension(std::string param) : is_unknown(false), is_int(false), dim(-1), param(std::move(param)) {} // NOLINT
-  Dimension(int64_t dim) : is_unknown(false), is_int(true), dim(dim) {} // NOLINT
+struct DimensionIR final {
```
Comment (xadupre) [inline onnx/common/ir.h:70]: Why rename it?

**Pull request closed without merging**